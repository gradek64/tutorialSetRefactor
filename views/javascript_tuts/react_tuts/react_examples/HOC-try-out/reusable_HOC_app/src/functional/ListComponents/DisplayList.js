import React from 'react';
import PropTypes from 'prop-types';

/*
  *@this is presentional component
  *@it is pure component that has no login just accepets data from props
*/

const RenderDisplayList = ({ list }) => (
  <div>
    {list.map(char => (
      <div key={char.name}>
        <div>Character: {char.name}</div>
        <div>Side: {char.side}</div>
      </div>
    ))}
  </div>
);
RenderDisplayList.propTypes = {
  list: PropTypes.instanceOf(Array).isRequired,
};

export default RenderDisplayList;
