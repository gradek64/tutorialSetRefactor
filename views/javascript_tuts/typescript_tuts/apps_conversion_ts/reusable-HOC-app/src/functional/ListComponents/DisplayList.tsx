import React from 'react';
import PropTypes from 'prop-types';

/*
  *@this is presentional component
  *@it is pure component that has no login just accepets data from props
*/
interface IProps {
  list:Array<{name:string,side:string}>
}

const RenderDisplayList:React.FC<IProps> = ({ list }) => (
  <div>
    {list.map(char => (
      <div key={char.name}>
        <div>Character: {char.name}</div>
        <div>Side: {char.side}</div>
      </div>
    ))}
  </div>
);
/* RenderDisplayList.propTypes = {
  list: PropTypes.instanceOf(Array).isRequired,
}; */

export default RenderDisplayList;
