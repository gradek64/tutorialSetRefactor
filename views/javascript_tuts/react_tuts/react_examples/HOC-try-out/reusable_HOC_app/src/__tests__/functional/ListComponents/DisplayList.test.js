import React from 'react';
import { shallow } from 'enzyme';

// component for snapshot test;
import DisplayList from '../../../functional/ListComponents/DisplayList';

let wrapper;
beforeAll(() => {
  wrapper = shallow(<DisplayList list={
    [
      { name: 'Greg', side: 'dark' },
      { name: 'Greg', side: 'light' },
    ]
  }
  />);
});

test('renders correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

// test props
test('list has to render side:dark or side:light', () => {
  const elDomTree = wrapper.debug();
  const hasSideDark = /dark/gi.test(elDomTree);
  const hasSideLigth = /light/gi.test(elDomTree);
  expect(hasSideDark && hasSideLigth).toBe(true);
});
