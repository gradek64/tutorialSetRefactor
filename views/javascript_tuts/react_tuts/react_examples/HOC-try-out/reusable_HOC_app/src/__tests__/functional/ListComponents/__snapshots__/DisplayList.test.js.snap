// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    key="Greg"
  >
    <div>
      Character: 
      Greg
    </div>
    <div>
      Side: 
      dark
    </div>
  </div>
  <div
    key="Greg"
  >
    <div>
      Character: 
      Greg
    </div>
    <div>
      Side: 
      light
    </div>
  </div>
</div>
`;
